src/golang/lib/execution_environment/zip.go

```diff
@@ -0,0 +1,81 @@
+package execution_environment
+
+import (
+	"archive/zip"
+	"bytes"
+	"io/ioutil"
+	"sort"
+	"strings"
+
+	"github.com/dropbox/godropbox/errors"
+)
+
+const (
+	ReqFileName           = "requirements.txt"
+	PythonVersionFileName = "python_version"
+)
+
+var (
+	ErrPythonVersionMissing    = errors.New("Python version file is missing.")
+	ErrRequirementsFileMissing = errors.New("Requirement file is missing.")
+)
+
+func InferDependenciesFromZipFile(zipball []byte) (*ExecutionEnvironment, error) {
```

Comment:
@likawind Let's change this function name to `ExtractDependenciesFromZipFile`, because there is no actual inference going on here.

Reply:
Also, the name of this file (`zip.go`) sounds a bit vague. Can we use sth like `extract_dependencies.go`?